Sources/ChatGPTAPIManager/ChatGPTAPIManager.swift

```diff
@@ -206,6 +208,57 @@ final public class ChatGPTAPIManager {
     }
     
+    /// Requests audio transcription based on the provided parameters.
+    ///
+    /// - Parameters:
+    ///   - fileUrl: The URL of the audio file to be transcribed.
+    ///   - prompt: The prompt or context for the transcription.
+    ///   - temperature: (Optional) The temperature value for generating diverse transcriptions. Defaults to nil.
+    ///   - language: (Optional) The language of the transcription. Defaults to nil.
+    ///   - model: (Optional) The model to be used for transcription. Defaults to ChatGPTModels.
+    ///   - endPoint: The URL endpoint for the transcription service.
+    ///   - completion: A completion handler to be called with the result of the transcription request.
+    ///                 The handler takes a Result object, which contains either the transcribed text or an error.
+    ///                 Use the `.success` case to access the transcribed text and the `.failure` case to handle errors.
+    public func audioTranscriptionRequest(fileUrl: URL, prompt: String, temperature: Double? = nil, language: String? = nil, model: ChatGPTModels, endPoint: APPURL, completion: @escaping (Result<String, Error>) -> Void) {
+        
+        self.audioTranscription(fileUrl: fileUrl, prompt: prompt,temperature: temperature,language: language, model: model, endPoint: endPoint, completion: { result in
+            
+            switch result {
+            case.success(let success):
+                completion(.success(success))
+            case.failure(let error):
+                completion(.failure(error))
+            }
+            
+        })
+        
+    }
+    /// Requests audio translation based on the provided parameters.
+    ///
+    /// - Parameters:
+    ///   - fileUrl: The URL of the audio file to be translated.
+    ///   - prompt: (Optional) The prompt or context for the translation. Defaults to nil.
+    ///   - temperature: (Optional) The temperature value for generating diverse translations. Defaults to nil.
+    ///   - model: The ChatGPT model to use for translation.
+    ///   - endPoint: The endpoint URL for the API request.
+    ///   - completion: The completion block called with the result of the request. The block receives a Result object containing either the translated text as a String in case of success, or an Error in case of failure.
+    public func audioTranslationRequest(fileUrl: URL, prompt: String? = nil, temperature: Double? = nil, model: ChatGPTModels, endPoint: APPURL, completion: @escaping (Result<String, Error>) -> Void) {
+        // Implementation of audio translation request goes here
+        self.audioTranslation(fileUrl: fileUrl,prompt: prompt,temperature: temperature, model: model, endPoint: endPoint, completion: { result in
+            
+            switch result {
+            case.success(let success):
+                completion(.success(success))
+            case.failure(let error):
+                completion(.failure(error))
+            }
```

Comment:
If there is no processing involved: pass public method completion into private method.